README.md

```diff
@@ -23,10 +23,14 @@ Mistral.rs is a fast LLM inference platform supporting inference on a variety of
 
 `cargo run --release --features ... -- -i plain -m meta-llama/Meta-Llama-3-8B-Instruct -a llama`
 
+`nix run github:EricLBuehler/mistral.rs -- -i plain -m meta-llama/Meta-Llama-3-8B-Instruct -a llama`
+
 **Running the new Phi 3 model with 128K context window**
 
 `cargo run --release --features ... -- -i plain -m microsoft/Phi-3-mini-128k-instruct -a phi3`
 
+`nix run github:EricLBuehler/mistral.rs -- -i plain -m microsoft/Phi-3-mini-128k-instruct -a phi3`
```

Comment:
Can we specify features and ensure that there is a release build here?

Reply:
Nix is a tool for reproducible builds and deployments. You define a build and it does the same thing every single time. Here's what the build command looks like now:

`cargo build --release --message-format json-render-diagnostics --locked`

It still uses cargo under the hood to compile the project.

This specific flake uses [ipetkov/crane](https://github.com/ipetkov/crane) to build this. [Here's what the crane library provides as build overrides](https://github.com/ipetkov/crane/blob/master/lib/buildPackage.nix#L49).

This means that for each feature you could have a different package definition with a different `cargoExtraArgs` field specifying the feature set and a default package enabling all the features. 

You don't have the option to specify build arguments through the command line. This is something nix users have learned to live with.